### DISCUSSION TITLE
[BP#73] Ne se connecter à une base données que si nécessaire

### DISCUSSION THREAD

**Discussion by ddebarbieux** [cnumr/best-practices#111] [Bonnes pratiques] [answered]
L'exemple de cette bonne pratique est basé sur un CMS.

-> Ce n'est pas vraiment lié aux base de données
-> Cela dépend du CMS

**Comment by florinesueur** [comment]
Personnellement, je comprends de deux façons différentes cette bonne pratique qui se traduira par :
- Éviter d'utiliser une base de données quand ce n'est pas nécessaire. Exemple : un petit site vitrine avec la publication d'un nouvel article tous les mois, je proposerai du [flat-file](https://en.wikipedia.org/wiki/Flat-file_database).
- Éviter de requêter la base de données quand ce n'est pas utile et nécessaire. Exemple : une donnée non utile, exclure (comme dans l'exemple de la BP), préciser le demande de recherche avec des critères, ...

**Reply by ACTLEM** [reply]
Je comprends la remarque de @LGnord car l'exemple n'est pas le meilleur. Je préfère ceux de @florinesueur. Donc plutôt pour changer l'exemple ;-)

**Reply by tbroyer** [reply]
L'hypothèse de la connexion coûteuse n'est-elle pas limitée par les pools de connexion ?

L'exemple semble plus porter sur l'utilisation de cette connexion avec une requête, et est due à l'architecture de l'application / du site, et principalement le choix des URLs. Si toutes les resources statiques et publiques (ou toutes les resources dynamiques ou à accès restreint, bref, qui font appel à du code applicatif) sont dans une "branche" (`/public/`, `/static/`, `/assets/`, etc.) alors il devient assez facile de gérer le cas des resources absentes sans faire appel à la base de données "au cas où en fait ce serait une resource dynamique".
J'imagine que même pour des resources statiques mais avec accès restreint il serait possible de faire la vérification d'accès uniquement après vérification d'existence de la resource (mais attentions aux effets de bord sur la sécurité : si le serveur répond différemment ou avec un temps de réponse significativement différent, alors on peut déterminer si une resource existe mais est protégée, ou bien n'existe pas ; ça peut devenir une vulnérabilité exploitable ; tout dépend de la sensibilité des données)

**Comment by ACTLEM** [comment, marked as answer]
Création de l'Issue: https://github.com/cnumr/best-practices/issues/247
